### ISSUE TITLE
Consul Ingress

### ISSUE THREAD

**Issue by iamaverrick** [hashicorp/consul-helm#510]
we are using this consul helm chart for our consul deployment on our production environment. i know that the UI is enabled but i see no ingress associated with this chart. 

my question is how can i set an ingress for this the UI service using this chart. we use nginx ingress and would like to specify the host to use for the UI service.

**Comment by hilalhmuhammed** [comment]
@iamaverrick Add an ingress resource pointing to port 80 of the UI service and allowing path '/' under templates directory and run helm upgrade. The ingress resource is not packed with current official chart.

**Comment by iamaverrick** [comment]
@hilalhmuhammed thank you for reply but i have tried and still no luck this is my ingress config

```
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: consul
  namespace: cloud
  annotations:
    kubernetes.io/ingress.class: "nginx"
    nginx.org/redirect-to-https: "True"
    ingress.kubernetes.io/ssl-redirect: "True"
    external-dns.alpha.kubernetes.io/hostname: "consul.company.com"
spec:
  rules:
    - host: consul.company.com
    - http:
        paths:
        - path: /
          backend:
            serviceName: consul-consul-ui
            servicePort: 80
```

i have verified that consul is working properly i have deployed using the following 

helm upgrade --install consul hashicorp/consul --set global.datacenter=company --version 0.23.1 -n company

**Comment by ashwin-venkatesh** [comment]
Related to #130

**Comment by himmakam** [comment]
Hi, When I enable ingress for this, I get 503 service Temporarily Unavailable. Could someone please let me know how to fix this issue.

**Comment by thisisnotashwin** [comment]
Addressed by #774 

If Ingress is used, the ingress performs the TLS termination so you have to ensure that either the Ingress controller can perform SSL-passthrough and that it is enabled, or that the http port of the UI is open (httpsOnly is false)
